pallets/polkadex-ido/src/lib.rs

```diff
@@ -521,316 +242,135 @@ pub mod pallet {
-                        T::AssetManager::create(token_a.into(), team.clone(), true, 1)?;
-                        T::AssetManager::mint_into(token_a.into(), &round_account_id, amount.saturated_into())?;
-                    }
-                    _ => {
-                        return Err(<Error<T>>::MintNativeTokenForbidden.into());
-                    }
-                }
-            } else {
-                Self::transfer(token_a, &team, &round_account_id, amount.saturated_into())?;
-                //ensure!(.is_ok(), <Error<T>>::TransferTokenAFromTeamAccountFailed);
-            }
+            Self::transfer(token_a, &team, &round_account_id, amount.saturated_into())?;
             <InfoFundingRound<T>>::insert(round_id, funding_round);
             <InfoProjectTeam<T>>::insert(team, round_id);
             Self::deposit_event(Event::FundingRoundRegistered(round_id));
             Ok(())
         }
 
-        /// Project team whitelists investor for the given round for the given amount.
+        /// Invest in a funding round
         ///
         /// # Parameters
         ///
         /// * `round_id`: Funding round id
-        /// * `investor_address`: Investor
-        /// * `amount`: The max amount that investor will be investing in tokenB
+        /// * `amount`: BalanceOf<T>
         #[pallet::weight((10_000, DispatchClass::Normal))]
-        pub fn whitelist_investor(origin: OriginFor<T>, round_id: T::Hash, investor_address: T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
-            let team: T::AccountId = ensure_signed(origin)?;
-            ensure!(!<InfoFundingRound<T>>::contains_key(&round_id), Error::<T>::FundingRoundNotApproved);
-            ensure!(<WhitelistInfoFundingRound<T>>::contains_key(&round_id), Error::<T>::FundingRoundDoesNotExist);
-            let funding_round = <WhitelistInfoFundingRound<T>>::get(round_id).ok_or(Error::<T>::FundingRoundNotApproved)?;
-            ensure!(team.eq(&funding_round.creator), <Error<T>>::NotACreater);
-            ensure!(<InfoInvestor<T>>::contains_key(&investor_address), <Error<T>>::InvestorDoesNotExist);
-            let current_block_no = <frame_system::Pallet<T>>::block_number();
-            ensure!(current_block_no < funding_round.close_round_block && current_block_no >= funding_round.start_block, <Error<T>>::NotAllowed);
-            <WhiteListInvestors<T>>::insert(round_id, investor_address.clone(), amount);
-            Self::deposit_event(Event::InvestorWhitelisted(round_id, investor_address));
-            Ok(())
-        }
-
-        /// Investor claiming for a particular funding round.
-        ///
-        /// # Parameters
-        ///
-        /// * `round_id`: Funding round id
-        #[pallet::weight((10_000, DispatchClass::Normal))]
-        pub fn claim_tokens(origin: OriginFor<T>, round_id: T::Hash) -> DispatchResult {
+        pub fn invest(origin: OriginFor<T>, round_id: T::Hash, amount: BalanceOf<T>) -> DispatchResult {
```

Comment:
`origin` is not included in the docbook because it's not an argument added by us, it's like boilerplate code for extrinsic, all extrinsics should have this argument.